Stop using distutils (removed in Python 3.12) to locate the Python 3
site-packages directory. Also teach the bundled FindPython3Interp module
about current CPython versions and the unversioned "python" / "python3"
names.

diff -Nru libproxy-0.4.18.orig/bindings/python/python3/CMakeLists.txt libproxy-0.4.18/bindings/python/python3/CMakeLists.txt
--- libproxy-0.4.18.orig/bindings/python/python3/CMakeLists.txt	2022-06-20 15:54:49.000000000 +0200
+++ libproxy-0.4.18/bindings/python/python3/CMakeLists.txt	2026-09-07 19:32:18.061734505 +0200
@@ -14,7 +14,7 @@
 
     execute_process(COMMAND
 		    ${PYTHON3_EXECUTABLE}
-		    -c "import distutils.sysconfig ; print (distutils.sysconfig.get_python_lib(plat_specific=0))"
+		    -c "import sysconfig; print(sysconfig.get_path('purelib'))"
 		    OUTPUT_VARIABLE PYTHON3_SITEPKG_DIR
 		    OUTPUT_STRIP_TRAILING_WHITESPACE)
   endif()
diff -Nru libproxy-0.4.18.orig/cmake/FindPython3Interp.cmake libproxy-0.4.18/cmake/FindPython3Interp.cmake
--- libproxy-0.4.18.orig/cmake/FindPython3Interp.cmake	2022-06-20 16:06:50.000000000 +0200
+++ libproxy-0.4.18/cmake/FindPython3Interp.cmake	2026-09-07 19:32:18.061819326 +0200
@@ -56,7 +56,7 @@
 
 unset(_Python3_NAMES)
 
-set(_Python3_VERSIONS 3.11 3.10 3.9 3.8 3.7 3.6 3.5 3.4 3.3 3.2 3.1 3.0)
+set(_Python3_VERSIONS 3.16 3.15 3.14 3.13 3.12 3.11 3.10 3.9 3.8 3.7 3.6 3.5 3.4 3.3 3.2 3.1 3.0)
 
 if(Python3Interp_FIND_VERSION)
     if(Python3Interp_FIND_VERSION_COUNT GREATER 1)
@@ -78,6 +78,7 @@
         set(_PYTHON3_FIND_OTHER_VERSIONS ${_PYTHON${Python3Interp_FIND_VERSION_MAJOR}_VERSIONS})
     endif()
 else()
+    list(APPEND _Python3_NAMES python python3)
     set(_PYTHON3_FIND_OTHER_VERSIONS ${_PYTHON3_VERSIONS})
 endif()
 find_program(PYTHON3_EXECUTABLE NAMES ${_Python3_NAMES})
